UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Hotspot.cs

```diff
@@ -11,14 +15,269 @@ public class Hotspot
+					node.Position, LayerType.Effects, OverlayType.FireFusion);
+
+				node.ReactionManager.TileChangeManager.RemoveOverlaysOfType(
+					node.Position, LayerType.Effects, OverlayType.FireRainbow);
+			}
+
+			//Despawn firelight
+			if (firelight != null)
+			{
+				_ = Despawn.ServerSingle(firelight.gameObject);
+			}
+		}
+
 		public void Process()
 		{
-			Reactions.React(node.GasMix, node.PositionMatrix);
+			UpdateColour();
 		}
+
+		private void UpdateColour()
+		{
+			if(firelight == null) return;
+
+			var temperature = node.GasMix.Temperature;
+			var temp2Colour = Temp2Colour(temperature);
+			var heatR = temp2Colour.r * 255;
+			var heatG = temp2Colour.g * 255;
+			var heatB = temp2Colour.b * 255;
+			var heatA = 255f;
+
+			var greyscaleFire = 1f; //This determines how greyscaled the fire is.
+
+			//This is where fire is very orange, we turn it into the normal fire texture here.
+			if (temperature < 5000)
+			{
+				var normalAmt = DMMath.GaussLerp(temperature, 1000, 3000);
+				heatR = DMMath.Lerp(heatR, 255, normalAmt);
+				heatG = DMMath.Lerp(heatG, 255, normalAmt);
+				heatB = DMMath.Lerp(heatB, 255, normalAmt);
+				heatA -= DMMath.GaussLerp(temperature, -5000, 5000) * 128;
+				greyscaleFire -= normalAmt;
+			}
+
+			//Past this temperature the fire will gradually turn a bright purple
+			if (temperature > 40000)
+			{
+				var purpleAmt = temperature <  DMMath.Lerp(40000, 200000, 0.5f) ? DMMath.GaussLerp(temperature, 40000, 200000) : 1;
+				heatR = DMMath.Lerp(heatR, 255, purpleAmt);
+			}
+
+			//Somewhere at this temperature nitryl happens.
+			if (temperature > 200000 && temperature < 500000)
+			{
+				var sparkleAmt = DMMath.GaussLerp(temperature, 200000, 500000);
+				var newColour = new Color(1f, 1f, 1f, sparkleAmt);
+
+				var currentColour = node.ReactionManager.TileChangeManager.GetColourOfFirstTile(node.Position,
+					OverlayType.FireSparkles, LayerType.Effects);
+
+				//Only add/remove if we need to
+				if (currentColour != newColour)
+				{
+					if (currentColour != null && hasSparkle)
+					{
+						//Remove old so it can be replaced by one with different alpha value
+						node.ReactionManager.TileChangeManager.RemoveOverlaysOfType(
+							node.Position, LayerType.Effects, OverlayType.FireSparkles);
+					}
+
+					hasSparkle = true;
+
+					//Add new
+					node.ReactionManager.TileChangeManager.AddOverlay(
+						node.Position, TileType.Effects, "FireSparkles", color: newColour);
+				}
+			}
+			else if (hasSparkle)
+			{
+				hasSparkle = false;
+
+				//Remove as its not needed anymore
+				node.ReactionManager.TileChangeManager.RemoveOverlaysOfType(
+					node.Position, LayerType.Effects, OverlayType.FireSparkles);
+			}
+
+			//Lightning because very anime.
+			if (temperature > 400000 && temperature < 1500000)
+			{
+				if (hasOvercharge == false)
+				{
+					hasOvercharge = true;
+
+					//Add new
+					node.ReactionManager.TileChangeManager.AddOverlay(
```

Comment:
my new problem is temperature changes quite a bit with fire reactions so it might send a lot of messages?

Reply:
*my only problem

Reply:
could add an internal timer to do it only every couple seconds?

Reply:
humm, theoretically the best would be custom message that is really highly compact tho I suppose we'll see if it's a problem then sort it out then